src/instructlab/configuration.py

```diff
@@ -380,8 +380,8 @@ class _train(BaseModel):
 
     deepspeed_cpu_offload_optimizer: bool
 
-    lora_rank: int
-    lora_quantize_dtype: str
+    lora_rank: Optional[int] = None
```

Comment:
You'll want to use `int | None` here - Pydantic doesn't seem to interpret `typing.Optional` properly:

```suggestion
    lora_rank: int | None = None
```

Reply:
The default `None` is not set in the config properly if set this way, but since this is more readable I kept this suggestion with a small change to be able to set a default.